desktop/core/src/desktop/api2.py

```diff
@@ -924,37 +924,45 @@ def gist_create(request):
   '''
   Only supporting Editor App currently.
   '''
-  response = {'status': 0}
 
   statement = request.POST.get('statement', '')
   gist_type = request.POST.get('doc_type', 'hive')
   name = request.POST.get('name', '')
   description = request.POST.get('description', '')
 
-  if not name:
-    name = _('%s Query') % gist_type.capitalize()
+  response = _gist_create(request.get_host(), request.is_secure(), request.user, statement, gist_type, name)
+
+  return JsonResponse(response)
+
+
+def _gist_create(host_domain, is_http_secure, user, statement, gist_type, name=''):
```

Comment:
Nice!

This now makes it trivial to test and share within the app!

Reply:
Agreed!